sdk/src/main/java/io/opentelemetry/sdk/resources/Resource.java

```diff
@@ -114,7 +114,25 @@ public static Resource getTelemetrySdk() {
    */
   public static Resource create(Attributes attributes) {
     checkAttributes(Objects.requireNonNull(attributes, "attributes"));
-    return new AutoValue_Resource(attributes);
+    Attributes.Builder attrBuilder = Attributes.newBuilder();
+    attributes.forEach(new NullValueFilter(attrBuilder));
+    return new AutoValue_Resource(attrBuilder.build());
+  }
+
+  private static final class NullValueFilter implements KeyValueConsumer<AttributeValue> {
+    private final Attributes.Builder attrBuilder;
+
+    private NullValueFilter(Attributes.Builder attrBuilder) {
+      this.attrBuilder = attrBuilder;
+    }
+
+    @Override
+    public void consume(String key, AttributeValue value) {
+      if (value.getType() == AttributeValue.Type.STRING && value.getStringValue() == null) {
+        return;
+      }
```

Comment:
Why other values like null array is not removed?

Reply:
I think along these lines, if we're using `null` to model deletion, we should just store `null` directly, not an attribute value with a value of `null`.

Reply:
Because null values inside arrays are meaningful. Since the `Attributes` API does not create `null` attributes, it cannot exist a null value but only a value that contains an `AttributeValue` with null. Does it make sense?

Reply:
No, it does not make sense. What if the "array" is null, not the value inside the array?

Reply:
The "array" cannot be null. `Resource` uses the `Attributes` class for storing attributes which creates always an object for arrays: https://github.com/open-telemetry/opentelemetry-java/blob/master/api/src/main/java/io/opentelemetry/common/Attributes.java#L204

Reply:
Very strange behavior for arrays then:
https://github.com/open-telemetry/opentelemetry-java/blob/78ed6490ba47568f5241604ea34f18d3ae6f3c4a/api/src/main/java/io/opentelemetry/common/AttributeValue.java#L331

We replace with empty array/list.... Not too much consistency.

Reply:
General Java guidelines are to never return null for a collection, but use an empty one (per Josh Bloch in Effective Java). So, I think this behavior is a good one.

Reply:
We discussed about this. It is not we return null, it is we get null and add an entry with empty collection, why then not add an entry with empty string?

Reply:
I have uniformed the handling of `null` in 18e3719. The following cases are dropped (or removed if the key was already present):
- null strings, e.g. `(String) null`
- new arrays with 0 length, e.g. `new String[0]`
- null arrays, e.g. `(String[]) null`

Instead, the following are stored as attributes:
- Empty strings
- Arrays with null values as members, e.g. `new String[] {null}`